monitoring/timedrift_test.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"context"
+	"net"
+	"time"
+
+	"github.com/gravitational/satellite/agent"
+	"github.com/gravitational/satellite/agent/health"
+	"github.com/gravitational/satellite/agent/proto/agentpb"
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+	"github.com/gravitational/trace"
+	"github.com/gravitational/ttlmap"
+	"github.com/jonboulle/clockwork"
+	"github.com/sirupsen/logrus"
+
+	serf "github.com/hashicorp/serf/client"
+	"gopkg.in/check.v1"
+)
+
+type TimeDriftSuite struct {
+	c     *timeDriftChecker
+	clock clockwork.Clock
+}
+
+var _ = check.Suite(&TimeDriftSuite{})
+
+func (s *TimeDriftSuite) SetUpSuite(c *check.C) {
+	s.c = &timeDriftChecker{self: node1}
+	s.clock = clockwork.NewFakeClock()
+}
+
+func (s *TimeDriftSuite) TestTimeDriftChecker(c *check.C) {
+	tests := []struct {
+		Comment string
+		Nodes   []serf.Member
+		Times   map[string]time.Time
+		Result  []*pb.Probe
+	}{
+		{
+			Comment: "Time drift to both nodes is acceptable",
+			Nodes:   []serf.Member{node1, node2, node3},
+			Times: map[string]time.Time{
+				node2.Name: s.clock.Now().Add(100 * time.Millisecond),
+				node3.Name: s.clock.Now().Add(200 * time.Millisecond),
+			},
+			// Since we're using frozen time, latency will be 0 so
+			// time drift will be exactly the amount we specified.
+			Result: []*pb.Probe{
+				s.c.successProbe(node2, 100*time.Millisecond),
+				s.c.successProbe(node3, 200*time.Millisecond),
+			},
+		},
+		{
+			Comment: "Time drift to both nodes is acceptable, one node is lagging behind",
+			Nodes:   []serf.Member{node1, node2, node3},
+			Times: map[string]time.Time{
+				node2.Name: s.clock.Now().Add(100 * time.Millisecond),
+				node3.Name: s.clock.Now().Add(-100 * time.Millisecond),
+			},
+			// Since we're using frozen time, latency will be 0 so
+			// time drift will be exactly the amount we specified.
+			Result: []*pb.Probe{
+				s.c.successProbe(node2, 100*time.Millisecond),
+				s.c.successProbe(node3, -100*time.Millisecond),
+			},
+		},
+		{
+			Comment: "Time drift to node-3 is high",
+			Nodes:   []serf.Member{node1, node2, node3},
+			Times: map[string]time.Time{
+				node2.Name: s.clock.Now().Add(100 * time.Millisecond),
+				node3.Name: s.clock.Now().Add(500 * time.Millisecond),
```

Comment:
In this case, `500*time.Millisecond` vs `100 * time.Millisecond` does not tell me much, it would've been more readable with:
```go
node2.Name: acceptableDrift(),
node2.Name: driftOverThreshold(),
```